pypiserver/__init__.py

```diff
@@ -114,6 +114,15 @@ def backwards_compat_kwargs(kwargs: dict, warn: bool = True) -> dict:
     return updated_kwargs
 
 
+def _get_health_endpoint(config: RunConfig, app: Bottle) -> str:
+    """Verify the health_endpoint and fallback to default if invalid."""
+    # Avoid health check route uses exist endpoints
+    for rule in (route.rule for route in app.routes if route.rule != "/"):
+        if config.health_endpoint.startswith(rule):
+            return DEFAULTS.HEALTH_ENDPOINT
```

Comment:
That's a suggestion👍

When route conflicts it will [raise RuntimeError](https://github.com/pypiserver/pypiserver/pull/442/commits/55fa396999c07be76ed58cfe4b14d860809368a1), so the [DEFAULTS is no longer needed](https://github.com/pypiserver/pypiserver/pull/442/commits/d2849d8a48f499628d80d7e015bf6e119cbe7220) during [setup_routes_from_config()](https://github.com/pypiserver/pypiserver/pull/442/commits/f1ea6707739e4720c93430df114926e11c03a731)